### PR TITLE
Document `caml_result` in the manual (FFI chapter)

### PR DESCRIPTION
**Pull request by gasche**
This PR is a companion to #13013, providing documentation for the new `caml_result` type and convention in the OCaml manual, FFI chapter.

I mostly removed the documentation of the previous encoded-exceptions API, except for in-passing mentions that it may still be used for compatibility with pre-5.3 codebases. People who need to understand the encoded-exception API will have to look for older versions of the manual.

### CODE REVIEW

File: manual/src/cmds/intf-c.etex
Comment (ghost) [inline manual/src/cmds/intf-c.etex]:
```suggestion
exceptions, and "Result_unit" is a convenient alias for
```

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: `at` -> `has` ? Maybe make two sentences ?

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: that returns _a value_ of type

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: `"CAMLlocal1(bar)"` -> `"CAMLlocal1(foo)"`

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: result returning C functions have their name suffixed by `_res`

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: is available since OCaml 5.3

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: which are of type

**Comment by gasche** [comment]
Thanks @dustanddreams and @dbuenzli, I took your comments into account. I also actually checked that the example compiles (not bad, I only wrote `caml_alloc2` instead of `caml_alloc_2`) and used an itemized list for a more structure presentation of the `caml_result` API.

File: manual/src/cmds/intf-c.etex
Comment (dbuenzli) [inline manual/src/cmds/intf-c.etex]: Sorry it seems that this comment didn't make it to my review: these examples are not typical but rather blueprints (a typical example would involve a ressource). Also in general in docs use present tense. So:

Some typical usage examples would be as follows: -> Some examples of use:

Comment (gasche) [reply manual/src/cmds/intf-c.etex]: Done, thanks!

File: manual/src/cmds/intf-c.etex
Comment (NickBarnes) [inline manual/src/cmds/intf-c.etex]: s/fuction/function/

**Review by NickBarnes — APPROVED** [review]
Looks good to me (apart from that one typo).

File: manual/src/cmds/intf-c.etex
Comment (gasche) [reply manual/src/cmds/intf-c.etex]: (Grr, it looks like the 'n' key on my keyboard is getting tired somehow, I made such typos fifty times in the last few days.) Thanks!

File: manual/src/cmds/intf-c.etex
Comment (gadmm) [inline manual/src/cmds/intf-c.etex]: "reraising" sounds confusing: either it raises into OCaml (which was not done yet), or it has been "raised" as a caml_result (in which case we don't get that it raises into OCaml).

File: manual/src/cmds/intf-c.etex
Comment (gadmm) [inline manual/src/cmds/intf-c.etex]: "OCaml function"

File: manual/src/cmds/intf-c.etex
Comment (gadmm) [inline manual/src/cmds/intf-c.etex]: result-returning

**Review by gadmm — APPROVED** [review]
This looks good to me.

You could also add a link to the section ss:c-result in the last paragraph of the section ss:c-exceptions which mentions the caml_exception_* functions.

**Comment by ghost** [comment]
Shouldn't this get merged soon since #13013 has been?

**Comment by gasche** [comment]
I would certainly be in favor and/but this requires the approval of a maintainer.

(I also need to do another round on the feedback of @gadmm -- thanks! -- but this seems orthogonal to approval.)

File: manual/src/cmds/intf-c.etex
Comment (MisterDA) [inline manual/src/cmds/intf-c.etex]:
```suggestion
Alternatively, the function \verb"caml_process_pending_actions_res" is provided, that
```

I think that "variant" is already reserved and an alternative naming could be used.

**Comment by gasche** [comment]
I took the comments of @gadmm and @MisterDA into account (thanks!) and rebased the PR. This still needs a maintainer approval to be merged.

**Review by shindere — APPROVED** [review]

**Comment by shindere** [comment]
Just approved.

I didn't merge yet because I noticed a discrepancy between the reviewers
mentionned in the commit andthose mentionned in theChanges entry. I
don'tmind but wated to give an opportunity forthisto be fixed, if you
wanted so, @gashce. Feel free to merge whenever you like as far asI am
concerned.

**Comment by gasche** [comment]
Thanks @shindere! When I restarted working on this PR I had forgotten that it had its own Changes entry, so I used commit metadata (which I don't normally use) to credit reviewers. This is fixed now -- the Changes has all names, and I removed the Reviewed-by tag, nothing against them but they are not standard practice in OCaml land so there is little point in using them just once.

**Pull request merged**
